File: JavaScript_Courses/JavaScript_Basics/JS_BP_2022_10/02_Conditional_Statements_Exercise/03_Time_15_Minutes.js
function time15Minutes(input) {
    let hours = Number(input[0]);  // 1
    let minutes = Number(input[1]);

    let minutesPlus15 = minutes + 15;

    // console.log(minutesPlus15);

    if (minutesPlus15 >= 60) {
        hours += 1;  // hours = hours + 1;
        minutesPlus15 -= 60;  // minutesPlus15 = minutesPlus15 - 60;
    }

    if (hours == 24) {  // 0, 1, ... 23; 24 => 0
        hours = 0;
    }

    if (minutesPlus15 < 10) {  // (minutesPlus15 <= 9)
        console.log(`${hours}:0${minutesPlus15}`);  // 0, 1, 2,... 9
    } else {
        console.log(`${hours}:${minutesPlus15}`);
    }

}

// time15Minutes(["1", "46"]);
// time15Minutes(["23", "55"]);

File: Python_Courses/Python_Basics/Python_PB_2025_03/02_Conditional_Statements_Exercise/03_Time_15_Minutes.py
# 1.
hours = int(input())
minutes = int(input())

# 2.
minutes += 15  # minutes = minutes + 15
if minutes >= 60:  # 71
    minutes -= 60  # 11
    hours += 1  # 1 -> 2

if hours >= 24:
    hours = 0

# 3.
print(f'{hours}:{minutes:02d}')
